node/tests/SharedTests.ts

```diff
@@ -3526,6 +3526,39 @@ export function runBaseTests<Context>(config: {
         },
         config.timeout,
     );
+
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        `dbsize test_%p`,
+        async (protocol) => {
+            await runTest(async (client: BaseClient) => {
+                // flush all data
+                expect(await client.customCommand(["FLUSHALL"])).toBe("OK");
+
+                // check that DBSize is 0
+                expect(await client.dbsize()).toBe(0);
+
+                // set 10 random key-value pairs
+                for (let i = 0; i < 10; i++) {
+                    const key = `{key}:${uuidv4()}`;
+                    const value = "0".repeat(Math.random() * 7);
+
+                    expect(await client.set(key, value)).toBe("OK");
+                }
+
+                // check DBSIZE after setting
+                expect(await client.dbsize()).toBe(10);
+
+                // additional test for the cluster client
+                if (client instanceof GlideClusterClient) {
+                    expect(await client.customCommand(["FLUSHALL"])).toBe("OK");
+                    const key = uuidv4();
+                    expect(await client.set(key, "value")).toBe("OK");
+                    expect(await client.dbsize("allPrimaries")).toBe(1);
```

Comment:
allPrimaries is the same as the default route. It should be "primary" (or whatever the Node client's equivalent is) to align with the Java test (see https://github.com/valkey-io/valkey-glide/pull/1540/files#diff-7ab084eac47fc1e2de7c47594968e6a070001320de1e2dffe1a8e40bd8125840).

Reply:
https://github.com/valkey-io/valkey-glide/blob/1138e48962765e20d2327cd6653305e2f9758b2d/node/src/GlideClusterClient.ts#L105 not sure if there is such an option currently